lib/rubocop/cop/rspec/receive_messages.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for multiple messeages stubbed on the same object.
+      #
+      # @example
+      #   # bad
+      #   before do
+      #     allow(Service).to receive(:foo).and_return(bar)
+      #     allow(Service).to receive(:baz).and_return(qux)
+      #   end
+      #
+      #   # good
+      #   before do
+      #     allow(Service).to receive_messages(foo: bar, baz: qux)
+      #   end
+      #
+      class ReceiveMessages < Base
+        extend AutoCorrector
+        include RangeHelp
+
+        MSG = 'Use `receive_messages` instead of multiple stub on line(s) ' \
+              '%<loc>s.'
+
+        # @!method allow_receive_message?(node)
+        def_node_matcher :allow_receive_message?, <<-PATTERN
+          (send (send nil? :allow ...) :to (send (send nil? :receive ...) ...) ...)
+        PATTERN
+
+        # @!method allow_argument(node)
+        def_node_matcher :allow_argument, <<-PATTERN
+          (send (send nil? :allow (const nil? $_) ...) ...)
+        PATTERN
+
+        # @!method receive_node(node)
+        def_node_search :receive_node, <<-PATTERN
+          $(send (send nil? :receive ...) ...)
+        PATTERN
+
+        # @!method receive_and_return_argument(node)
+        def_node_matcher :receive_and_return_argument, <<-PATTERN
+          (send (send nil? :allow ...) :to (send (send nil? :receive (sym $_)) :and_return (send nil? $_)))
```

Comment:
What if the argument to `and_return` is a literal, and not a `send`?

Reply:
It should be searchable as well. I updated this PR.